src/System.Private.Xml/src/System/Xml/Dom/XmlName.cs

```diff
@@ -182,18 +183,9 @@ public static int GetHashCode(string name)
             int hashCode = 0;
             if (name != null)
             {
-                unchecked
-                {
-                    for (int i = name.Length - 1; i >= 0; i--)
-                    {
-                        char ch = name[i];
-                        if (ch == ':') break;
-                        hashCode += (hashCode << 7) ^ ch;
-                    }
-                    hashCode -= hashCode >> 17;
-                    hashCode -= hashCode >> 11;
-                    hashCode -= hashCode >> 5;
-                }
+                string[] splitted = name.Split(":");
+                ReadOnlySpan<byte> bytes = MemoryMarshal.AsBytes(splitted[splitted.Length - 1].AsSpan());
```

Comment:
the old code didn't allocate (`string.Split`).
```
name.AsSpan(name.LastIndexOf(':')) ?
```

Reply:
colon shouldn't included

Reply:
@buyaa-n - this cannot allocate - this is being called hundreds of times

Reply:
Maybe wait until https://github.com/dotnet/corefx/issues/26528 is merged? Would not require allocating for split @krwq

Reply:
There's no need to Split here at all.